cmd/common/registry.go

```diff
@@ -228,17 +228,30 @@ func buildCommandFromGadget(
 
 				// This kind of gadgets return directly the result instead of
 				// using the parser
-				result, err := runtime.RunGadget(gadgetCtx)
+				results, err := runtime.RunGadget(gadgetCtx)
 				if err != nil {
 					return fmt.Errorf("running gadget: %w", err)
 				}
 
-				transformed, err := transformResult(result)
-				if err != nil {
-					return fmt.Errorf("transforming result: %w", err)
+				for node, result := range results {
+					transformed, err := transformResult(result.Payload)
```

Comment:
Shouldn't this check `result.Error` before calling `transformResult`?

Reply:
Yeah, and the error check above that needs to be moved down to allow partial results.